// Copyright 2024 The Cross-Media Measurement Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package experimental.dp_consistency.src.main.proto.reporting;

option java_package = "experimental.dp_consistency.src.main.proto.reporting";
option java_multiple_files = true;
option java_outer_classname = "MeasurementsProto";

message MeasurementDetail {
  string measurement_policy = 1;
  string set_operation = 2;
  bool is_cumulative = 3;
  repeated string data_providers = 4;

  message MeasurementResult {
    int64 reach = 1;
    double standard_deviation = 2;
    string metric = 3;
  }
  repeated MeasurementResult measurement_results = 5;
}

message ReportSummary {
  repeated MeasurementDetail measurement_details = 1;
}
